util.js

```diff
@@ -159,3 +159,50 @@ exports.metaBackup = (msgValue, property) => {
 
   return msgValue.meta
 }
+
+exports.AsyncJobQueue = class AsyncJobQueue {
```

Comment:
Someone must have made a module for this already?

Reply:
yes, this started out as a minimal refactor and as I added tests got more complex.
Given it's working and working (I think) exactly like the code which was previously there, I'm kinda keen to leave this as is.

Feel free anyone to drop an alternative in and see if it passes the same tests!